tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_enrollment.py

```diff
@@ -137,27 +137,35 @@ def run_test(id, number_of_tests_per_thread):
     print("Test parameters:")
     print("- target: https://%s:%s" % (args.hostname, args.port))
 
-    number_of_threads = args.number_of_clients
-    print("- number of clients: %s" % number_of_threads)
+    number_of_clients = args.number_of_clients
+    print("- number of clients: %s" % number_of_clients)
 
-    number_of_tests_per_thread = args.number_of_tests_per_client
-    print("- number of tests per client: %s" % number_of_tests_per_thread)
+    number_of_tests_per_client = args.number_of_tests_per_client
+    print("- number of tests per client: %s" % number_of_tests_per_client)
 
-    # create the specified number of threads
-    threads = []
+    # Create the specified number of threads
     cert_list = []
     issuance_times = []
     start = timer()
 
-    for t in range(1, number_of_threads + 1):
-        logger.info("Starting client %s" % t)
-        t1 = threading.Thread(target=run_test, args=(t, number_of_tests_per_thread))
-        t1.start()
-        threads.append(t1)
+    clients = []
+    for i in range(number_of_clients):
+        id = i + 1
+        # Create a PKIConnection object that stores the details of the CA.
+        connection = PKIConnection('https', args.hostname, args.port, cert_paths=args.ca_cert_path)
+
+        # The pem file used for authentication. Created from a p12 file using the
+        # command -
+        # openssl pkcs12 -in <p12_file_path> -out /tmp/auth.pem -nodes
+        connection.set_authentication_cert(args.client_cert)
+        client = TestClient( id, connection, number_of_tests_per_client)
+        clients.append(client)
+        client.start()
```

Comment:
I'd suggest to split the client setup into a separate loop so we can measure just the actual test:
```
for i in range(...):
    connection = PKIConnection(...)
    client = TestClient(....)
    clients.append(client)

start = timer()

for client in clients:
    client.start()

for client in clients:
    client.join()

end = timer()
```

Reply:
Fixed for both the scripts